### PR TITLE
[HOPSWORKS-2578] Create onlinefs etc directory under data dir

### PR DESCRIPTION
**Pull request by kouzant**
https://logicalclocks.atlassian.net/browse/HOPSWORKS-2578

### CODE REVIEW

**Review by berthoug — APPROVED** [review]

File: attributes/default.rb [hunk 1]
```diff
@@ -12,6 +12,11 @@
 default['onlinefs']['logs']                   = "#{node['onlinefs']['home']}/logs"
 default['onlinefs']['token']                  = "#{node['onlinefs']['etc']}/token"
 
+# Data volume directories
+default['onlinefs']['data_volume']['root_dir']  = "#{node['data']['dir']}/onlinefs"
+default['onlinefs']['data_volume']['etc_dir']   = "#{node['onlinefs']['data_volume']['root_dir']}/etc"
```
Comment (SirOibaf) [inline attributes/default.rb:17]: `etc` is contains only configuration and a token that's regenerated every time. You don't need to move it to the `data_volume`

**Review by SirOibaf — CHANGES_REQUESTED** [review]

File: templates/default/onlinefs.service.erb [hunk 2]
```diff
@@ -6,6 +6,7 @@ Requires = docker.service
 [Service]
 ExecStart = /usr/bin/docker run --rm --name %n \
            -v <%= node['onlinefs']['home'] %>:/onlinefs \
+           -v <%= node['onlinefs']['data_volume']['root_dir']%>:<%= node['onlinefs']['data_volume']['root_dir']%> \
```
Comment (SirOibaf) [inline templates/default/onlinefs.service.erb:9]: Why is this necessary?

File: attributes/default.rb (hunk 1 above)
Comment (kouzant) [reply attributes/default.rb:17]: As of now, the token is not regenerated when you restore a backup in hopsworks.ai

File: templates/default/onlinefs.service.erb (hunk 2 above)
Comment (kouzant) [reply templates/default/onlinefs.service.erb:9]: We mount `node['onlinefs']['home']` under `/onlinefs` in the container, which contains `etc/` and `logs/` directories. These directories are symbolic links now and docker does not dereference the link when it mounts the volumes. If you don't mount the target directory, you end up with broken symlinks in the container.

**Review by SirOibaf — APPROVED** [review]
